### ISSUE TITLE
Issues with training a new net on the simplified formula.

### ISSUE THREAD

**Issue by gekkehenker** [nodchip/Stockfish#34]
Seems like I can't get the latest binaries working right with my older data. I've tried adjusting eta from 1.0 to 0.1, I've adjusted lambda from 1.0 to 0.5 and 0.3. nn_batch_size from 1000 to 10000.
Loss stays high and nets play horribly.

I think Additional_pylon has the same issues

JJosh luckily seems to be progressing with the simplified formula though.

Below is an example command I've used:

learn targetdir train loop 100 batchsize 1000000 use_draw_in_training 1 use_draw_in_validation 1 eta 0.1 lambda 0.3 eval_limit 32000 nn_batch_size 1000 newbob_decay 0.5 eval_save_interval 250000000 loss_output_interval 1000000 mirror_percentage 50 validation_set_file_name val\vald10.bin

The training data I've used is the same with which I managed to create my first strong net.

Anyone having similar issues?
I also still have to test "stockfish-nnue-2020-06-27-1" binaries.

**Comment by nodchip** [comment]
When I adjust the eta value, I sometimes try, 0.01, 0.02, 0.05, 0.1, 0.2, 0.5, 1.0, 2.0, 5.0 and 10.0.  I draw the graph of test_cross_entropy and move_accuracy, and select the best eta.  Is it possible to experiment with those values?

**Comment by gekkehenker** [comment]
I'll try a higher eta than 1.

**Comment by gekkehenker** [comment]
Just to be 100% sure.

The latest learn binaries are (fully) compatible with old trainingdata right?
Or do I need to generate new data?

**Comment by nodchip** [comment]
The latest learn binaries should be fully compatible with old training data.  Because I changed functions which are used only to calculate the gradients in SGD (stochastic gradient descent).

Could you please generate new training data, and compare the training results?

**Comment by gekkehenker** [comment]
Thanks for the answer.

I'll start generating new training data and try that.

**Comment by gekkehenker** [comment]
It's a seperate issue and fixed by modifying source code and compiling it myself. But I want to make a new user manual for NNUE.

How to use "use_draw_in_training" and "use_draw_in_validation" with the learn command?

Do I need to use the command "nn_options" with it? Do you have an example command where it's working correctly?

**Comment by nodchip** [comment]
> It's a seperate issue and fixed by modifying source code and compiling it myself. But I want to make a new user manual for NNUE.

Could you please let me know what was the issue?  If the fix is needed in my repository, could you please send a pull request?

> How to use "use_draw_in_training" and "use_draw_in_validation" with the learn command?

> learn targetdir train loop 100 batchsize 1000000 use_draw_in_training 1 use_draw_in_validation 1 eta 0.1 lambda 0.3 eval_limit 32000 nn_batch_size 1000 newbob_decay 0.5 eval_save_interval 250000000 loss_output_interval 1000000 mirror_percentage 50 validation_set_file_name val\vald10.bin

The example in your first comment is right.  Please add "use_draw_in_training 1" and "use_draw_in_validation 1" to a learn command.

**Comment by NightlyKing** [comment]
I have no problems  with it so far
